### PR TITLE
[UI smoke test] Saved logins in pwa UI smoke test

### PR DESCRIPTION
**Pull request by AndiAJ**
New `saveLoginsInPWATest` UI smoke test.

✅  Successfully passed 60x on Firebase

<details>

<summary> 📈 Results (dropdown)</summary>

matrix | result | logs | details
-- | -- | -- | --
matrix-27l0pgjw16oaa | success | logs | 1 test cases passed
matrix-2zuy983odx9kx | success | logs | 1 test cases passed
matrix-1swus00fd30eg | success | logs | 1 test cases passed
matrix-k22y77vbsobva | success | logs | 1 test cases passed
matrix-uimvcds53vzza | success | logs | 1 test cases passed
matrix-16onlrtwmdh9a | success | logs | 1 test cases passed
matrix-1wighv5othf4p | success | logs | 1 test cases passed
matrix-21duj7cx5psd9 | success | logs | 1 test cases passed
matrix-4hal7fazeicla | success | logs | 1 test cases passed
matrix-1vj1eoi6afrsx | success | logs | 1 test cases passed
matrix-1b7mt5cbfkayo | success | logs | 1 test cases passed
matrix-gcpy3uqhjphoa | success | logs | 1 test cases passed
matrix-2nlya7tk6wzac | success | logs | 1 test cases passed
matrix-1rnhcvw4ia8y5 | success | logs | 1 test cases passed
matrix-42xfhrb9b6v0a | success | logs | 1 test cases passed
matrix-zxuwqt4z3ml0a | success | logs | 1 test cases passed
matrix-2u9rtdncoy3vj | success | logs | 1 test cases passed
matrix-117v4uhfue41i | success | logs | 1 test cases passed
matrix-1xbv7iq0j6x8a | success | logs | 1 test cases passed
matrix-35s61i5o37wh1 | success | logs | 1 test cases passed
matrix-14w2p992kk1dw | success | logs | 1 test cases passed
matrix-2etsg7juuqt7x | success | logs | 1 test cases passed
matrix-16m4dux3g8v40 | success | logs | 1 test cases passed
matrix-2wwpmpelkfy8n | success | logs | 1 test cases passed
matrix-3mrn6wlagahn9 | success | logs | 1 test cases passed
matrix-y064bwygnilla | success | logs | 1 test cases passed
matrix-17hj83otk6i1g | success | logs | 1 test cases passed
matrix-6z7k5p6reyz2a | success | logs | 1 test cases passed
matrix-1xekx126gw6zv | success | logs | 1 test cases passed
matrix-1axtfxal0qzc5 | success | logs | 1 test cases passed
matrix-1iitl0ayb4jb6 | success | logs | 1 test cases passed
matrix-3vz0iccwfxlr7 | success | logs | 1 test cases passed
matrix-3dsl1lcmsqmu8 | success | logs | 1 test cases passed
matrix-2huxpqe0pscw7 | success | logs | 1 test cases passed
matrix-506xe8sqx4era | success | logs | 1 test cases passed
matrix-2o1f5po7kxlqo | success | logs | 1 test cases passed
matrix-tgoxt7u0tczna | success | logs | 1 test cases passed
matrix-3qv1q3dfppjw2 | success | logs | 1 test cases passed
matrix-q7wgc6fjniyaa | success | logs | 1 test cases passed
matrix-ucvmnjkx9px0a | success | logs | 1 test cases passed
matrix-21x0gkg4cz459 | success | logs | 1 test cases passed
matrix-ro2q3aolkduca | success | logs | 1 test cases passed
matrix-3p2ktvcbsu3jg | success | logs | 1 test cases passed
matrix-1315p9qyktvnq | success | logs | 1 test cases passed
matrix-2oegwbgsfwp6z | success | logs | 1 test cases passed
matrix-7l3bw9q8d8soa | success | logs | 1 test cases passed
matrix-1bi3panpch1st | success | logs | 1 test cases passed
matrix-lx9a2qlecm0ia | success | logs | 1 test cases passed
matrix-39zon7nopxxwm | success | logs | 1 test cases passed
matrix-2yl5oj8rlznmx | success | logs | 1 test cases passed
matrix-1cdjnmsnbhl7p | success | logs | 1 test cases passed
matrix-2nsay8gn4ffda | success | logs | 1 test cases passed
matrix-2f2j105t485l4 | success | logs | 1 test cases passed
matrix-1cft7z996mg3w | success | logs | 1 test cases passed
matrix-1ayffxutot4om | success | logs | 1 test cases passed
matrix-1g58k3jz7acwj | success | logs | 1 test cases passed
matrix-3v1uyt2d7cx5q | success | logs | 1 test cases passed
matrix-3g15afvuikkwa | success | logs | 1 test cases passed
matrix-2l474n1eiczhp | success | logs | 1 test cases passed
matrix-nh9g3hwjf1gpa | success | logs | 1 test cases passed

</details>

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt [hunk 1]
```diff
@@ -517,6 +519,43 @@ class SettingsPrivacyTest {
             verifyEmptyHistoryView()
         }
     }
+
+    @SmokeTest
+    @Test
+    fun saveLoginsInPWATest() {
+        val pwaPage = "https://andiaj.github.io/testapp/loginForm"
+        val shortcutTitle = "yay app"
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt:527]: nit: let's just use "TEST_APP"

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt [hunk 2]
```diff
@@ -517,6 +519,43 @@ class SettingsPrivacyTest {
...
+            mDevice.waitForIdle()
+            fillAndSubmitLoginCredentials("mozilla", "firefox")
+            verifySaveLoginPromptIsDisplayed()
+            saveLoginFromPrompt("Save")
+            openAppFromExternalLink(pwaPage)
+
+            browserScreen {
+            }.openThreeDotMenu {
+            }.openSettings {
+                scrollToElementByText("Logins and passwords")
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt:545]: Is there a resource string for this?

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 3]
```diff
@@ -550,6 +550,94 @@ class BrowserRobot {
         }
     }
 
+    fun fillAndSubmitLoginCredentials(userName: String, password: String) {
+
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:554]: nit: remove newline

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 4]
```diff
@@ -550,6 +550,94 @@ class BrowserRobot {
...
+                    .className("android.widget.Button")
+                    .packageName("$packageName")
+            )
+
+        submitLoginButton.click()
+        mDevice.waitForIdle()
+    }
+
+    fun verifyPrefilledLoginCredentials(userName: String, shortcutTitle: String) {
+
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:594]: nit: remove newline

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 5]
```diff
@@ -550,6 +550,94 @@ class BrowserRobot {
...
+            )
+
+        submitLoginButton.click()
+
+        var currentTries = 0
+        while (currentTries++ < 3) {
+            try {
+                assertTrue(userNameTextBox.waitForExists(waitingTime))
+                break
+            } catch (e: AssertionError) {
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:624]: Is this necessary? Can you try this with `RetryTestRule`?

**Review by AaronMT — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt]: Done

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt]: Not applicable anymore

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt (hunk 4 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:578]: Done

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt (hunk 5 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:586]: Yes it is, the RetryTestRule will not work.
I need to check the pre-filled in the PWA, the RetryTestRule takes me back to Fenix.

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt (hunk 3 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt]: Done

**Comment by AndiAJ** [comment]
Made a couple of changes and everything seems to be in order, successfully passed 50x on Firebase

**Review by AaronMT — APPROVED** [review]

**Pull request merged**